docs/UserGuide.md

```diff
@@ -357,26 +345,25 @@ A lesson can be categorised into 2 types:
 1. A **weekly** recurring lesson
 2. A one-off makeup lesson.
 
-The essential fields for a lesson are:
-* Start date
-* Time range
-* Subject
-* Rate
-
-An optional field for both types of lesson is:
-* Homework
-* Outstanding Fees (refer to more details in this [section](#managing-lesson-fees))
+The essential fields for a lesson are indicated by the presence of the '*'. The rest of the fields are optional.
 
-Optional fields for a **recurring** lesson is:
-* End date
-* Cancelled dates
+Field | Prefix | Constraints | Examples | Remark|
+------------------|-------|-----------| --------| -----|
+*Start date |`date/`| Dates should be of the format dd MMM yyyy and adhere to the following constraints:<br/>1. dd and yyyy are numerical characters.<br/>2. MMM are alphabetical characters. e.g. Jan, Feb, ..., Dec<br/>3. Must be a valid date for the year."|`date/20 jan 2022`<br/>`date/4 nov 2000`| Dates are case-insensitive and allows single digit day without the leading zero.|
+*Time range        | `time/` | Lesson time range should be formatted as HHmm-HHmm and adhere to the following constraints:<br>1. Start time must be before end time.<br/>2. Lesson should be conducted between 8am and 10pm, inclusive.<br/> | `time/1100-1300`<br/>`time/2000-2130` | |
+*Subject          | `subject/` | Subject should only contain alphanumeric characters and spaces, and it should not be blank | `subject/Math`<br/> `subject/Language Arts` | |
+*Lesson rate      | `rates/` | Money-related fields such as lesson rate should be formatted with a decimal point '.' as a separator between the dollars and cents, and adhere to the following constraints:<br/>1. Lesson rates should only contain numbers and at most one decimal point.<br/>2. The lesson rate should not start or end with a decimal point and should have at most two decimal places. | `rates/37.50`<br/>`rates/40` | The lesson's rate refers to the fee of the lesson per hour. This rate will be used in the calculation of fees due after each lesson.<br/> |
```

Comment:
```suggestion
*Lesson rate      | `rates/` | We follow the [Singapore convention](https://www.dfa.cornell.edu/treasurer/cash-management/processinginternational/intl-currency) of using a decimal point '.' to separate dollars and cents, Lesson rates: <br/> 1. should only contain numbers, at most one decimal point, and two decimal places <br/>2. should not start or end with a decimal point | `rates/37.50`<br/>`rates/40` 
```
Note somewhere else that rate is $ per H so the table is neater

Reply:
Would it not be weird to separate the fields and the remarks for that field?